### ISSUE TITLE
Suggestion for referencing method

### ISSUE THREAD

**Issue by weallen** [cortex-lab/neuropixels#10]
I had previously been using the method Nick recommended of using an Ag/AgCl wire soldered to the headstage, and then placing the wire near the craniotomy, but I was having variable results depending on where the wire was placed relative to the craniotomy and how much saline was in the bath/how dry the agar became. (And fear of accidentally breaking the electrode often kept me from putting it too close...)

Inspired by some papers from the Svoboda lab, I've found that I get more consistent results by chronically implanting a platinum/iridium wire soldered to a pin connector
(https://www.a-msystems.com/p-189-male-miniature-pin-connector-fits-model-1800-3000-headstage-leads.aspx) either over the cerebellum or over visual cortex (not important for my task). 

A wire soldered to a female connector 
(https://www.a-msystems.com/p-188-female-miniature-pin-connector-fits-a-m-systems-electrodes.aspx)
connected to the headstage is then used to plug into the reference pin during each recording session. This also avoids having to have any sort of bath over the craniotomy, besides some saline to keep it moist.

Are there any downsides to this? Is there a significant advantage to referencing directly near the insertion site vs somewhere else in the brain?

**Comment by nsteinme** [comment]
No downsides that I know of to the skull screw or implanted pin strategy, so long as you are competent at performing that implant. I don't think it matters too much how you reference for most things: in fact for spiking activity you can leave the probe completely un-referenced, and perform common average referencing on the data and it will still be fine. For LFP, in most cases you it is probably ideal to use a current-source density or other kind of local re-referencing anyway. 

I don't think I understood the part about saline, though - the problem with saline and referencing is when it dries out, isn't it? But you're saying you still have to keep it from drying out? Fwiw, I have now been using agar covered by silicone oil, which is stable for as long as I do my recordings.

**Comment by weallen** [comment]
I realized that the variability was mostly actually due to where the reference electrode was placed relative to the craniotomy, and it always made me nervous positioning the reference wire next to the craniotomy on each experiment. (Although I think I was maybe using a thicker, more rigid wire than I needd to.)

Weirdly, even when I switched to internal referencing in SpikeGLX, it seemed to be sensitive to where the external reference was -- when I just left it dangling in their air, there was tons of noise...

**Comment by nsteinme** [comment]
Yes, internal referencing does not seem to work in Phase3A. They should
have fixed it in phase3B.

What variability was due to reference electrode placement? Variability in
what?

**Comment by weallen** [comment]
It seemed like it was picking up oscillatory noise when it was resting on the skull further away from the craniotomy and/or there was less saline in the well. This noise obviously totally messed up the LFP, and wasn't completely eliminated by CAR. 

Are there any other substantial changes for the Phase 3B probes? I remember Tim saying once that they were trying to make them more durable?

**Comment by nsteinme** [comment]
Yes, if the reference loses connection with the probe (as by drying out) then the probe will be un-referenced and pick up noise - I often saw what you describe when I wasn't using the agar/oil system, that at the moment of drying out there was an oscillatory noise that would grow and grow. Btw I wouldn't probably recommend using CAR on the LFP band because some true signals may exist on much of the probe in the lower frequencies. 

Re: Phase3B, there is some small change to the geometry of the place where the shank meets the base, intended to make them more robust to breakage there. I don't think I have a complete list, but other changes I know about are small: fewer internal reference sites (but adding a big one at the tip); some small modifications to the gain/filter settings that are available; a different site pattern will be available (vertically aligned sites rather than checkerboard. checkerboard also still available). I believe they will now have the Janelia-developed dovetail holder-system, but not 100% sure when that will be implemented.

**Comment by PhantomSpike** [comment]
Hi Will and Nick,

I use a system for referencing very similar to what @weallen  describes where I have a male/female crimp connectors. I use these ones ([male](https://uk.rs-online.com/web/p/circular-connector-contacts/6876364/) and [female](https://uk.rs-online.com/web/p/circular-connector-contacts/6876361/)) and I am quite happy with them. On one end it connects to the headstage ground and on the other to the reference. The reference is a simple Ag wire which I chlorinate.  I usually insert the ref in the brain in some place not important for my auditory experiments e.g. Cerebellum or PFC. As long as the ref is in the brain I am quite happy with the signal typically <12 uV during recordings. However, this is still a bit awkward and I would strongly prefer if I could use internal referencing. Let me know if it would be helpful to anybody and I can send some pictures. :)

@nsteinme , do you know approximately when the Phase 3B probes with the internal referencing working will be released?

I also have another question regarding the craniotomy and keeping the brain and reference happy.

Currently I use agar and flush it with saline every few hours. Agar is tricky for me as I am afraid to not burn the brain or make it too hard so that it breaks the probe. What concentration do people normally use (I go for ~ 1.25% weight/volume)? I also usually dissolve the agar in PBS rather than pure water as I think this should increase conductivity and help with the reference issue. 

@nsteinme  in your setup, do you fist apply the agar, then the silicon oil and then insert the probe? Do you keep it moist at all in any way or you just let it be? How long does the prep usually last for?

Cheers,

Alex

**Comment by nsteinme** [comment]
Phase 3B probes will be the ones on sale this summer. There are supposed to
be test versions available a few months ago but I haven't seen any.

I use similar concentration of agar as you, dissolved in cortex buffer.

Yes, that's the order I use (agar -> si oil -> probe), and then I just let
it be. It's stable for 2 hours, which is as long as my experiments ever go.
